wordpress-mobile/WordPress-Android
# Prepublishing Nudges : extract publish settings fragment & viewmodel

Fixes #11727

## Findings
`EditPostPublishSettingsFragment` & `EditPostPublishSettingsViewModel` contains the logic used by the `Post Settings` screen to modify the publish scheduling settings. To maintain code reliability and maintainability they were utilized. 

## Solution
1. `EditPostPublishSettingsFragment` & `EditPostPublishSettingsViewModel` were made abstract so that both the Prepublishing Nudges & Post Settings could reuse the functionality. From an architecture standpoint the new abstract `PublishSettingsFragment` contains logic that allows the subclasses to implement their own layout, set a fragment type for the time & date pickers, and the ability to additional view setup i.e for the Prepublishing fragment, we needed to wire up the custom toolbar. 

To test:
1. Create a post. 
2. Go to Post Settings. 
3. Click on Publish. 
4. Change the Time & Date. 
5. Also, change the Notification. 
6. Go back and ensure the values have been set and reflected in the Post Settings UI. 

### Review 

1. Only 1 reviewer is needed but anyone can review.
2. Ensure that #11634 is merged.
3. Set this base branch to `feature/prepublishing_nudges`
4. Remove `Not Ready for Merge`
5. Merge :)


PR submission checklist:
